p2p/net/mock/ratelimiter.go

```diff
@@ -0,0 +1,63 @@
+package mocknet
+
+import (
+	"time"
+)
+
+type ratelimiter struct {
+	bandwidth    float64       // bytes per nanosecond
+	allowance    float64       // in bytes
+	maxAllowance float64       // in bytes
+	lastUpdate   time.Time     // when allowance was updated last
+	count        int           // number of times rate limiting was applied
+	duration     time.Duration // total delay introduced due to rate limiting
+}
+
+func NewRatelimiter(bandwidth float64) *ratelimiter {
```

Comment:
how is this supposed to work / be used, @whyrusleeping ? i think an example is in order